datafusion/core/src/datasource/listing/table.rs

```diff
@@ -214,33 +214,6 @@ impl ListingTableConfig {
     }
 }
 
-#[derive(Debug, Clone)]
-///controls how new data should be inserted to a ListingTable
-pub enum ListingTableInsertMode {
-    ///Data should be appended to an existing file
-    AppendToFile,
-    ///Data is appended as new files in existing TablePaths
-    AppendNewFiles,
-    ///Throw an error if insert into is attempted on this table
-    Error,
```

Comment:
It is worth highlighting this also removes ListingTableInsertMode::Error, I wasn't sure that this was important to preserve but happy to restore if people feel strongly. @devinjdangelo

Reply:
Removing error is fine by me. A formal access control mechanism would be better than the error mode option.

I do think there is some value to append mode for ListingTable. Specifically it enables tables backed by a single file where inserts append to that file.

This is a relatively niche usecase so I'm not bothered if it is removed.

Reply:
> Specifically it enables tables backed by a single file where inserts append to that file.

I believe this use-case is now covered by StreamTable unless I am missing something here?

Reply:
That could be (I haven't dug into StreamTable yet). The usecase for append I'm thinking of isn't really streaming related. More like a batch job that appends to one file or a fixed set of files on disk.

Reply:
StreamTable will do the former, but cannot, yet, handle multiple files. We could add this but I would rather scream test if anyone is actually doing this

Reply:
It would be good to hear about a usecase of appending to a set of existing files -- it is not a usecase I am familiar with but also my background is in analytic database where it is more common to make new files (and then "compact/rewrite" them as subsequent step) rather than modifying existing files